### PR TITLE
Refactor toolbox item change event

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #4203

### Proposed Changes

Creates dedicated class (`Blockly.Events.ToolboxItemSelect`) for toolbox item selects.

### Reason for Changes

See issue, but generally motivated by:
- better typing
- clearer events

### Test Coverage

Tested by logging events on playground and observing that expected events were logged.

Tested on:
 * Desktop Chrome

### CODE REVIEW

File: core/events_toolbox_select.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2018 Google LLC
```
Comment (alschmiedt) [inline core/events_toolbox_select.js:3]: Update to 2020

File: core/events_toolbox_select.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+goog.provide('Blockly.Events.ToolboxItemSelect');
+
+goog.require('Blockly.Events');
+goog.require('Blockly.Events.Ui');
+goog.require('Blockly.registry');
+goog.require('Blockly.utils.object');
+
+/**
+ * Class for a trashcan open event.
```
Comment (alschmiedt) [inline core/events_toolbox_select.js:21]: Update comment.

**Review by alschmiedt — APPROVED** [review]

File: core/events_toolbox_select.js (hunk 1 above)
Comment (moniika) [reply core/events_toolbox_select.js]: Done.

File: core/events_toolbox_select.js (hunk 2 above)
Comment (moniika) [reply core/events_toolbox_select.js]: Done.

**Pull request merged**